model-optimizer/extensions/middle/ONNXResize10ToInterpolateV4.py

```diff
@@ -0,0 +1,139 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging as log
+import numpy as np
+
+from extensions.ops.Cast import Cast
+from extensions.ops.activation_ops import Floor
+from extensions.ops.elementwise import Add, Mul
+from extensions.ops.interpolate import Interpolate
+from mo.front.common.layout import get_depth_dim, get_height_dim, get_width_dim
+from mo.front.common.partial_infer.utils import int64_array, float_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.passes.convert_data_type import data_type_str_to_np
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.const import Const
+from mo.ops.shape import Shape
+from mo.ops.strided_slice import StridedSlice
+
+
+def replace_resize(graph: Graph, resize: Node):
```

Comment:
We can perform this transformation during the front phase using operations `Rank`, `Range` etc.

Reply:
Done.

Reply:
Formally speaking, we can perform this transformation at the front stage, but we will get non-constant input `axes` of the `Interpolate` in this case. Hence, we will get the error 
```sh
Input 'axes' should be Constant.
```
from the nGraph infer function of `Interpolate-4`.

Reasons of the error is the following: 
1. we get non-Constant input `axes`, because MO generates IR without constant folding of operations `Range` and `Rank`;
2. the infer function of the nGraph operation `Interpolate-4` cannot infer output shape, if the input `axes` is non-Constant, and it is a behavior accordingly the specification of `Interpolate-4` (see https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/image/Interpolate_4.md);
3. in the front stage of MO, we don't know any shapes, and, therefore, we cannot calculate the `axes` input as a `Constant`.

Hence, we cannot really perform this transformation in the front stage.

Reply:
Fixed the shape infer function of the nGraph operation `Interpolate-4` to support for a non-constant input `axes`.